### PR TITLE
Upgrade django-rest-framework to `v3`

### PR DESCRIPTION
**Pull request by Ian-Foote**

### CODE REVIEW

File: user_management/api/tests/test_views.py [hunk 1]
```diff
@@ -318,6 +319,9 @@ def test_send_email(self):
         self.assertIn('auth/password_reset/confirm/', sent_mail.body)
         self.assertIn('https://', sent_mail.body)
 
+    # max_length is incorrectly missing from options, fixed in:
+    # https://github.com/tomchristie/django-rest-framework/commit/c8609ba652e1752e690c9e27e02b3531589d0c2c
```
Comment (Ian-Foote) [inline user_management/api/tests/test_views.py]: This should appear in `v3.0.6` of `django-rest-framework`.

File: user_management/api/serializers.py [hunk 2]
```diff
@@ -31,6 +38,7 @@ class Meta:
 
 
 class RegistrationSerializer(ValidateEmailMixin, serializers.ModelSerializer):
+    email = serializers.EmailField(validators=[unique_email_validator])
```
Comment (KevinEtienne) [inline user_management/api/serializers.py]: maybe the `email` field can be defined in the mixin? (Same apply to `UserSerializerCreate`)

Comment (Ian-Foote) [inline user_management/api/serializers.py]: This doesn't work - I think it's something to do with the metaclass `Serializer` uses.

**Comment by Ian-Foote** [comment]
@incuna/backend Review?

File: user_management/api/serializers.py [hunk 3]
```diff
@@ -8,18 +8,24 @@
...
+class UniqueEmailValidator(validators.UniqueValidator):
+    def filter_queryset(self, value, queryset):
+        """Check lower-cased email is unique."""
+        return super(UniqueEmailValidator, self).filter_queryset(
+            value.lower(),
+            queryset,
+        )
 
-        try:
-            User.objects.get_by_natural_key(email)
```
Comment (KevinEtienne) [inline user_management/api/serializers.py]: By removing `get_by_natural_key` project using `django-user-management` and modifying the email before being stored in the database would need to subclass the new `validator` to bring back `get_by_natural_key`.

Comment (Ian-Foote) [inline user_management/api/serializers.py]: Do you think we should continue to support that use-case directly here or should such projects handle it themselves? If we should support it here, can you add or sketch a test for this?

Comment (KevinEtienne) [inline user_management/api/serializers.py]: Yes, it feels natural for project/other library to handle it

**Comment by Ian-Foote** [comment]
@incuna/backend Any more comments?

**Pull request merged**